### PR TITLE
[CXF] Add CXF test templates & Bean-Validation-API support

### PR DESCRIPTION
**Pull request by jfiala**
* added the CXF client dependency (as this allows to run unittests as well inside the generated server)
* added test case templates
* also added Bean-Validation-API support for CXF (see #2549)

In fact, the current JavaCXFServerCodegen is exactly identical to the CXF client.

Only adding implementation stubs would make it to a the server project.

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaCXFServerCodegen.java [hunk 1]
```diff
@@ -38,6 +38,16 @@ public JavaCXFServerCodegen()
 
 
 	@Override
+	public void processOpts()
+	{
+		super.processOpts();
+		
+		supportingFiles.clear(); // Don't need extra files provided by AbstractJAX-RS & Java Codegen
+        writeOptional(outputFolder, new SupportingFile("pom.mustache", "", "pom.xml"));
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaCXFServerCodegen.java]: Alignment

**Comment by jfiala** [comment]
added full CXF server support including Spring + Swagger2Feature + WADL (WAR-file)

**Comment by wing328** [comment]
@jfiala first of all, my apologies for not reviewing this earlier. I did a test locally but got the following error:
```
Results :

Failed tests: 
  JavaClientOptionsTest>AbstractOptionsTest.checkOptionsHelp:44 These options weren't checked: useBeanValidation.

Tests run: 514, Failures: 1, Errors: 0, Skipped: 0
```
If you need help fixing it, please let me know.

Am I understanding correctly that this is a breaking change? If yes, please file it against `2.3.0` branch instead.

Please also rebase on the latest master resolve the merge conflicts. If you need help resolving the merge conflicts, I can help out as well.

**Comment by jfiala** [comment]
I fixed the AbstractOptionsTest and added the default false to the JavaClientOptionsProvider. So the change should be non-breaking, only if users activate the CLI-option useBeanvalidation they'll show up.

I also tried rebasing, but unfortunately and get this error message:
git rebase origin/master
First, rewinding head to replay your work on top of it...
error: The following untracked working tree files would be overwritten by checkout:
        modules/swagger-codegen/src/main/resources/TypeScript-Angular/api.d.mustache
        modules/swagger-codegen/src/main/resources/TypeScript-Angular/api.mustache
        modules/swagger-codegen/src/main/resources/TypeScript-Angular/model.mustache
        modules/swagger-codegen/src/main/resources/TypeScript-node/api.mustache
Please move or remove them before you can switch branches.
Aborting
could not detach HEAD

Can you give me a hint how I can resolve this?

**Comment by jfiala** [comment]
Regarding verifying non-breaking changes: So far it seems to me there are only tests for the behaviour of the languages, but not for the actual generator output. Don't you think it would be nice to have such contract tests to warrant the behaviour of Swagger-Codegen only changes if supposed to (such tests are e.g. in the Swagger2Markup project, it should be quite easy to set them up & keep them up to date using the generator and then the question of breaking changes should be a matter of the past...)?

**Comment by jfiala** [comment]
@wing328 if it is messy to pull the PR in, I can also reset my whole forked repo and redo all the changes once again against the current master ... (with all the beanvalidation + cxf server changes).

**Comment by jfiala** [comment]
Thx to this "git config core.ignorecase true" I was able to do the rebase for the branch jaxrs_cxf_2017 (new 0 commits behind / 43 commits ahead of master). 

I noted two things:
1.) JavaJaxRS/cxf/api.mustache: 
Replaced hard-coded "Response" with dynamic "{{#returnType}}{{{returnType}}} {{/returnType}}{{^returnType}}"

2.) DefaultCodegen.toRegularExpression():
DefaultCodegen.toRegularExpression() adds slashes to the pattern.
However, for Beanvalidation-patterns the slashes are not needed. 
I override toRegularExpression() in AbstractJavaCodegen.java as a temporary fix to make BeanValidation work, but I'm not sure why the slashes have been added?

**Comment by wing328** [comment]
@jfiala seems like the rebase failed as the PR now contains 250+ commits...

If it's not possible to submit a new (and clean) one, then I think the only option left is to cherry-pick the changes you've made.

**Comment by jfiala** [comment]
closed to create a new clean PR.

**Comment by jfiala** [comment]
The new PR contains the rebased changes ready for merging.

It seems the old PR was unable to pick them up (they've displayed clean in the branch view: https://github.com/swagger-api/swagger-codegen/compare/master...jfiala:jaxrs_cxf_2017).

**Pull request closed without merging**